policy/handler.go

```diff
@@ -115,13 +121,65 @@ func (h *Handler) Run(ctx context.Context, evalCh chan<- *Evaluation) {
 			h.updateHandler(currentPolicy, &p)
 			currentPolicy = &p
 		case <-h.ticker.C:
+
+			// Timestamp the invocation of this evaluation run. This can be
+			// used when checking cooldown or emitting metrics to ensure some
+			// consistency.
+			curTime := time.Now().UTC().UnixNano()
+
 			eval, err := h.generateEvaluation(currentPolicy)
 			if err != nil {
 				h.log.Error(err.Error())
 				return
 			}
-			if eval != nil {
-				evalCh <- eval
+
+			// If the evaluation is nil there is nothing to be done this time
+			// around.
+			if eval == nil {
+				continue
+			}
+
+			// If the target status includes a last event meta key, check for cooldown
+			// due to out-of-band events. This is also useful if the Autoscaler has
+			// been re-deployed.
+			if ts, ok := eval.TargetStatus.Meta[targetpkg.MetaKeyLastEvent]; ok {
+
+				// Convert the last event string. If an error occurs, just log and
+				// continue with the evaluation. A malformed timestamp shouldn't mean
+				// we skip scaling.
+				lastTS, err := strconv.ParseUint(ts, 10, 64)
+				if err != nil {
+					h.log.Error("failed to parse last event timestamp as uint64", "error", err)
+				} else {
```

Comment:
completely **personal** preference, but `} else {` should be avoided.

In this case, it can be avoided (and along with it, the indentation) by either using a `label` to skip down to ´evalCh`  or have `isInCooldown` return false if `lastTS.IsZero()` is `true` - or spin out the entire logic block here into a function - it's doing quite a lot of branching and indentation already :)